packages/strapi-connector-mongoose/lib/buildQuery.js

```diff
@@ -79,14 +79,15 @@ const buildQuery = ({
   searchParam,
   populate = [],
   aggregate = false,
+  session = null,
 } = {}) => {
   const search = buildSearchOr(model, searchParam);
 
   if (!hasDeepFilters(filters.where) && aggregate === false) {
-    return buildSimpleQuery({ model, filters, search, populate });
+    return buildSimpleQuery({ model, filters, search, populate, session });
```

Comment:
Can you pass with session in a second argument buildSImpleQuery({}, { session }) to differentiate parameter and query context. 

You can do the same for the other functions :)

Reply:
Sure, Done!